### PR TITLE
Allow to set the tag to use when cloning apiextensions.

### PR DESCRIPTION
**Pull request by oponder**
I added cobra and the `apiExtensionsTag` flag, so that we can control from which commit in the apiextensions repo this tool will build docs from.

Example, locking it to `v0.2.0`

```
$ go build && ./crd-docs-generator --apiExtensionsTag v0.2.0
/tmp/gitclone/docs/crd/application.giantswarm.io_app.yaml
/tmp/gitclone/docs/crd/application.giantswarm.io_appcatalog.yaml
/tmp/gitclone/docs/crd/core.giantswarm.io_ignition.yaml
/tmp/gitclone/docs/crd/infrastructure.giantswarm.io_awscluster.yaml
/tmp/gitclone/docs/crd/infrastructure.giantswarm.io_awsmachinedeployment.yaml
/tmp/gitclone/docs/crd/release.giantswarm.io_release.yaml
```

Example taking it from master currently, which has a CRD that is missing some properties
that this tool expects:

```
$ go build && ./crd-docs-generator --apiExtensionsTag master
/tmp/gitclone/docs/crd/application.giantswarm.io_app.yaml
/tmp/gitclone/docs/crd/application.giantswarm.io_appcatalog.yaml
/tmp/gitclone/docs/crd/core.giantswarm.io_ignition.yaml
/tmp/gitclone/docs/crd/infrastructure.giantswarm.io_awscluster.yaml
/tmp/gitclone/docs/crd/infrastructure.giantswarm.io_awscontrolplane.yaml
/tmp/gitclone/docs/crd/infrastructure.giantswarm.io_awsmachinedeployment.yaml
/tmp/gitclone/docs/crd/infrastructure.giantswarm.io_cluster.yaml
/tmp/gitclone/docs/crd/infrastructure.giantswarm.io_machinedeployment.yaml
/tmp/gitclone/docs/crd/provider.giantswarm.io_awsconfig.yaml
/tmp/gitclone/docs/crd/release.giantswarm.io_release.yaml
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x157c92b]

goroutine 1 [running]:
main.WriteCRDDocs(0xc000274dc0, 0x169b090, 0x8, 0x0, 0x0)
	/Users/oponder/code/giantswarm/crd-docs-generator/main.go:248 +0xbab
main.generateCrdDocs(0x7ffeefbff978, 0x6)
	/Users/oponder/code/giantswarm/crd-docs-generator/main.go:349 +0x2e9
main.main.func1(0xc0001b2000, 0xc000119860, 0x0, 0x2)
	/Users/oponder/code/giantswarm/crd-docs-generator/main.go:307 +0x37
github.com/spf13/cobra.(*Command).execute(0xc0001b2000, 0xc00010a190, 0x2, 0x2, 0xc0001b2000, 0xc00010a190)
	/Users/oponder/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001b2000, 0xc000119828, 0x16a011d, 0x10)
	/Users/oponder/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/oponder/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
	/Users/oponder/code/giantswarm/crd-docs-generator/main.go:315 +0x142
```

Also tried to make the actual error handling (so when it's not a panic case) more giant swarm like and print a stack trace.

```
$ go build && ./crd-docs-generator --apiExtensionsTag yolobolo
Error: execution error: Could not `git clone` source repository.
Tried to execute: /usr/bin/git clone -b yolobolo --depth 1 https://github.com/giantswarm/apiextensions.git /tmp/gitclone
exit status 128

--- Stack Trace ---
/Users/oponder/code/giantswarm/crd-docs-generator/main.go:328
/Users/oponder/code/giantswarm/crd-docs-generator/service/git/git.go:17
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,6 +11,7 @@ require (
 	github.com/huandu/xstrings v1.3.0 // indirect
 	github.com/mitchellh/copystructure v1.0.0 // indirect
 	github.com/shurcooL/sanitized_anchor_name v1.0.0 // indirect
+	github.com/spf13/cobra v0.0.5
```
Comment (marians) [inline go.mod:14]: Why not the latest version?

File: main.go [hunk 2]
```diff
@@ -10,15 +11,27 @@ import (
...
-	"github.com/giantswarm/crd-docs-generator/service/git"
+	"github.com/spf13/cobra"
 
 	"github.com/Masterminds/sprig"
 	"github.com/ghodss/yaml"
+	"github.com/giantswarm/crd-docs-generator/service/git"
 	"github.com/giantswarm/microerror"
 	blackfriday "gopkg.in/russross/blackfriday.v2"
 	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
 )
```
Comment (marians) [inline main.go:22]: The import from `github.com/giantswarm/crd-docs-generator`(internal package) should not be in the same group as external dependendcies. Also the empty line 15 should be removed.

File: main.go [hunk 3]
```diff
@@ -284,12 +297,35 @@ func WriteCRDDocs(crd *apiextensionsv1beta1.CustomResourceDefinition, outputFold
...
+		c := &cobra.Command{
+			Use:          "crd-docs-generator",
+			Short:        "crd-docs-generator is a command line tool for generating markdown files that document Giant Swarm's custom resources",
+			SilenceUsage: true,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return generateCrdDocs(crdDocsGenerator.apiExtensionsTag)
+			},
+		}
+
+		c.PersistentFlags().StringVar(&crdDocsGenerator.apiExtensionsTag, "apiExtensionsTag", "master", "The tag to use from the apiextensions repo when building crd documentation.")
```
Comment (marians) [inline main.go:313]:
```suggestion
		c.PersistentFlags().StringVar(&crdDocsGenerator.apiExtensionsTag, "apiextensions-version-tef", "master", "The git version reference (tag, branch, commit SHA) to use from the giantswarm/apiextensions repository")
```

1. Command line flags should be lowercase and use `-` as separator.
2. `master` is not a tag, and I'm assuming the flag will support all sorts of git version references.

File: main.go [hunk 4]
```diff
@@ -316,4 +352,20 @@ func main() {
...
+	json.Unmarshal([]byte(microerror.JSON(err)), &stackedError)
+
+	for i, j := 0, len(stackedError.Stack)-1; i < j; i, j = i+1, j-1 {
+		stackedError.Stack[i], stackedError.Stack[j] = stackedError.Stack[j], stackedError.Stack[i]
+	}
+
+	for _, entry := range stackedError.Stack {
+		fmt.Printf("%s:%d\n", entry.File, entry.Line)
+	}
 }
```
Comment (marians) [inline main.go:371]: Extra thanks for this effort to print a human-friendly stack trace!

File: go.mod (hunk 1 above)
Comment (oponder) [reply go.mod:14]: Not sure.. thats just what I got automatically somehow.

Comment (marians) [reply go.mod:14]: Can you try v0.0.7 manually?

File: main.go (hunk 3 above)
Comment (oponder) [reply main.go:313]: ok! i fixed a typo, and I think technically the thing we're referring to is a commit ref, not a version ref. Or maybe even just 'ref'

File: go.mod (hunk 1 above)
Comment (oponder) [reply go.mod]: Done 👍

File: main.go (hunk 3 above)
Comment (marians) [reply main.go]: Fine with me

**Pull request merged**